Dockerfile

```diff
@@ -1,16 +1,21 @@
-FROM amd64/alpine:3.18
+# syntax=docker/dockerfile:1
 
-ARG NAME=monaco
-ARG SOURCE=/build/${NAME}-linux-amd64
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS build
+WORKDIR /src
+ARG TARGETOS
+ARG TARGETARCH
+ARG VERSION=2.x
+COPY . .
+RUN GOOS=$TARGETOS GOARCH=$TARGETARCH CGO_ENABLED=0  go build -a -tags netgo -ldflags "-X github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/version.MonitoringAsCode=${VERSION} -w -extldflags '-static'" ./cmd/monaco
```

Comment:
Is there any really good reason to move building the binary into the docker container?
I get making each artifact build standalone that makes perfect sense - but making it standalone inside the docker builds feels like it just adds more overhead than benefit.

Couldn't we still build the correct binary inside the Jenkins pod by building it as part of the docker make target, or the Jenkins function `buildContainer`? 
Copying the sources and building inside Docker just seems like overhead and is probably slower than building outside.
Go should be able to build any target architecture on any architecture, doesn't it?

Reply:
Discussed in person: As we use buildx to build for different architectures in a single command, this is the easiest way to build the monaco binary for the correct architecture. Makes perfect sense to do it this way with that!